pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -535,6 +388,115 @@ func (d *DependenciesDistributor) removeScheduleResultFromAttachedBindings(bindi
 	return utilerrors.NewAggregate(errs)
 }
 
+func (d *DependenciesDistributor) createOrUpdateAttachedBinding(attachedBinding *workv1alpha2.ResourceBinding) error {
+	if err := d.Client.Create(context.TODO(), attachedBinding); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			klog.Infof("Failed to create resource binding(%s/%s): %v", attachedBinding.Namespace, attachedBinding.Name, err)
+			return err
+		}
+
+		existBinding := &workv1alpha2.ResourceBinding{}
+		key := client.ObjectKeyFromObject(attachedBinding)
+		if err := d.Client.Get(context.TODO(), key, existBinding); err != nil {
+			klog.Infof("Failed to get resource binding(%s/%s): %v", attachedBinding.Namespace, attachedBinding.Name, err)
+			return err
+		}
+
+		updatedBindingSnapshot := mergeBindingSnapshot(existBinding.Spec.RequiredBy, attachedBinding.Spec.RequiredBy)
+		existBinding.Spec.RequiredBy = updatedBindingSnapshot
+		existBinding.Labels = util.DedupeAndMergeLabels(existBinding.Labels, attachedBinding.Labels)
+		existBinding.Spec.Resource = attachedBinding.Spec.Resource
+
+		if err := d.Client.Update(context.TODO(), existBinding); err != nil {
+			klog.Errorf("Failed to update resource binding(%s/%s): %v", existBinding.Namespace, existBinding.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+// Start runs the distributor, never stop until stopCh closed.
+func (d *DependenciesDistributor) Start(ctx context.Context) error {
+	klog.Infof("Starting dependencies distributor.")
+	d.stopCh = ctx.Done()
+	resourceWorkerOptions := util.Options{
+		Name:          "resource detector",
+		KeyFunc:       detector.ClusterWideKeyFunc,
+		ReconcileFunc: d.reconcile,
+	}
+	d.EventHandler = fedinformer.NewHandlerOnEvents(d.OnAdd, d.OnUpdate, d.OnDelete)
+	d.Processor = util.NewAsyncWorker(resourceWorkerOptions)
+	d.Processor.Run(2, d.stopCh)
+	<-d.stopCh
+
+	klog.Infof("Stopped as stopCh closed.")
+	return nil
+}
+
+// SetupWithManager creates a controller and register to controller manager.
+func (d *DependenciesDistributor) SetupWithManager(mgr controllerruntime.Manager) error {
+	return utilerrors.NewAggregate([]error{
+		mgr.Add(d),
+		controllerruntime.NewControllerManagedBy(mgr).For(&workv1alpha2.ResourceBinding{}).
+			WithEventFilter(predicate.Funcs{
+				CreateFunc: func(event event.CreateEvent) bool {
+					bindingObject := event.Object.(*workv1alpha2.ResourceBinding)
+					if !bindingObject.Spec.PropagateDeps {
+						return false
+					}
+					if len(bindingObject.Spec.Clusters) == 0 {
```

Comment:
For newly created RBs, this condition is definitely not satisfied. Therefore, can we ignore the RB creation event?

Reply:
Here we mainly deal with the situation where only "create event" occurs after the controller-manager restarts.